src/include/network/protocol_handler.h

```diff
@@ -54,6 +55,7 @@ class ProtocolHandler {
 
   // The traffic cop used for this connection
   tcop::TrafficCop* traffic_cop_;
+  logging::WalLogManager* log_manager_;
```

Comment:
I think it is better to put log_manager_ as a member of traffic_cop. The log_manager is irrelated to protocol.

Reply:
Ok, and the callback register will be done the same way, only accessing the log_manager through the tcop, right?

Reply:
Yes